substrate/frame/support/src/traits/tokens/fungibles/regular.rs

```diff
@@ -489,18 +486,15 @@ pub trait Balanced<AccountId>: Inspect<AccountId> + Unbalanced<AccountId> {
 	fn pair(
 		asset: Self::AssetId,
 		amount: Self::Balance,
-	) -> (Debt<AccountId, Self>, Credit<AccountId, Self>) {
-		let total_issuance = Self::total_issuance(asset.clone());
-
-		// Maximum amount which can be issued.
-		let issue_cap = Self::Balance::max_value().saturating_sub(total_issuance);
-		// Maximum amount which can be rescinded.
-		let rescind_cap = total_issuance;
-
-		// Amount which can be both issued and recinded.
-		let capped_amount = amount.min(issue_cap).min(rescind_cap);
-
-		(Self::rescind(asset.clone(), capped_amount), Self::issue(asset, capped_amount))
+	) -> Result<(Debt<AccountId, Self>, Credit<AccountId, Self>), DispatchError> {
+		let issued = Self::issue(asset.clone(), amount);
+		let rescinded = Self::rescind(asset, amount);
+		if issued.peek() != rescinded.peek() {
```

Comment:
ditto

Reply:
it does not guarantee the entire amount is issued 

https://github.com/paritytech/polkadot-sdk/pull/1296/files/3d14f3d0a3dd3f807402f598602000795b01a8d5..c50c714c76cbbd2e3aacff8da4860def7fd5f794#diff-e04f7799b91b4225b1923b49d8484495f7868868ab32ed78919f310ffe268fd9L410-L411